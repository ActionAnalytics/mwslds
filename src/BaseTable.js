import React from 'react'

import { startCase } from 'lodash'

import './bcgov_bootstrap'
import withData from './DataLoader'

function getHeaderCols(row) {
  return Object.keys(row).map(key => startCase(key))
}

function getHeader(row) {
  return getHeaderCols(row).map(header => (
    <th key={header}>{header}</th>
  ))
}

function getBodyCols(row) {
  return <tr>{Object.values(row).map(val => <td>{val}</td>)}</tr>
}

function getBody(data) {
  return data.map(row => getBodyCols(row))
}

function BaseTable(props) {
  let header
  let body
  const data = props.data && props.data[props.dataValue]

  if (props.loading) {
    header = <th>Loading ...</th>
    body = <tr />
  } else if (data) {
    header = getHeader(data[0])
    body = getBody(data)
  } else {
    header = <th>No Data</th>
    body = <tr />
  }

  return (
    <table className="table table-striped table-hover table-bordered">
      <thead className="thead-dark">
        <tr>
          {header}
        </tr>
      </thead>
      <tbody>
        {body}
      </tbody>
    </table>
  )
}

export default withData(BaseTable, 'inspectors?regionCode=1')
