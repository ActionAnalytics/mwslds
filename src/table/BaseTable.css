
.table-header div {
  overflow: hidden !important;
  text-align: center;
}

.table .ReactVirtualized__Grid {
  border: 1px solid #ddd;
}

.table .ReactVirtualized__Grid__innerScrollContainer div {
  border-collapse: collapse;
  border-bottom: 1px solid #ddd;
  border-right: 1px solid #ddd;
  padding: 8px;
  line-height: 1.428;
}
