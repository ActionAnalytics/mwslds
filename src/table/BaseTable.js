import React from 'react'
import PropTypes from 'prop-types'

import { startCase } from 'lodash'

import '../style'

const propTypes = {
  data: PropTypes.array,
}

const defaultProps = {
  data: null,
}

function getHeaderCols(row) {
  return Object.keys(row).map(key => startCase(key))
}

function getHeader(row) {
  return getHeaderCols(row).map(header => (
    <th key={header}>{header}</th>
  ))
}

function getBodyCols(row) {
  const data = Object.keys(row).map((key) => {
    const val = row[key]
    let parsed = val
    if (val instanceof Object) {
      // this is pretty hacky it assumes all objects have either a code or an id
      parsed = val.code || val.id || ''
    }
    return <td key={key}>{parsed}</td>
  })
  // try to use the id as a key... if it has one
  const key = row.id || null
  return <tr key={key}>{data}</tr>
}

function getBody(data) {
  return data.map(row => getBodyCols(row))
}

function BaseTable(props) {
  const { data } = props
  let header
  let body

  if (data && data.length) {
    header = getHeader(data[0] || {})
    body = getBody(data)
  } else {
    header = <th>No Data</th>
    body = <tr />
  }

  return (
    <table className="table table-striped table-hover table-bordered">
      <thead className="thead">
        <tr>
          {header}
        </tr>
      </thead>
      <tbody>
        {body}
      </tbody>
    </table>
  )
}

BaseTable.propTypes = propTypes
BaseTable.defaultProps = defaultProps

export default BaseTable
