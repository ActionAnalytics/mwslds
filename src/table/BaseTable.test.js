import React from 'react'
import { configure, mount } from 'enzyme'
import { createSerializer } from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import BaseTable from './BaseTable'

configure({ adapter: new Adapter() })
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }))

describe('BaseTable', () => {
  it('renders', () => {
    const table = mount(<BaseTable />)
    expect(table).toMatchSnapshot()
  })

  it('renders with data', () => {
    const data = [
      { col1: 'aa', col2: 'ab', col3: 'ac' },
      { col1: 'ba', col2: 'bb', col3: 'bc' },
      { col1: 'ca', col2: 'cb', col3: 'cc' },
      { col1: 'da', col2: 'db', col3: 'dc' },
      { col1: 'ea', col2: 'eb', col3: 'ec' },
    ]
    const table = mount(<BaseTable keyField="col1" data={data} />)
    expect(table).toMatchSnapshot()
  })

  it('renders column labels', () => {
    const data = [
      { col1: 'aa', col2: 'ab', col3: 'ac' },
      { col1: 'ba', col2: 'bb', col3: 'bc' },
      { col1: 'ca', col2: 'cb', col3: 'cc' },
      { col1: 'da', col2: 'db', col3: 'dc' },
      { col1: 'ea', col2: 'eb', col3: 'ec' },
    ]
    const columnLabels = { col1: 'first', col2: 'second', col3: 'third' }
    const table = mount(<BaseTable keyField="col1" data={data} columnLabels={columnLabels} />)
    expect(table).toMatchSnapshot()
  })

  it('renders expand indicator column', () => {
    const data = [
      { col1: 'aa', col2: 'ab', col3: 'ac' },
    ]
    const table = mount(<BaseTable keyField="col1" data={data} expandComponent={() => <div />} />)
    expect(table).toMatchSnapshot()
  })

  it('renders expand compoent', () => {
    const data = [
      { col1: 'aa', col2: 'ab', col3: 'ac' },
    ]
    const update = () => {}
    const expand = jest.fn(() => <div />)
    mount(<BaseTable keyField="col1" data={data} expandComponent={expand} updateData={update} />)
    // make sure the expand component gets called with the row data and update function
    expect(expand.mock.calls[0][0]).toEqual(Object.assign({ updateTableData: update }, data[0]))
  })

  it('renders search compoent', () => {
    const data = [
      { col1: 'aa', col2: 'ab', col3: 'ac' },
    ]
    const filter = () => {}
    const search = jest.fn(() => <div />)
    mount(<BaseTable keyField="col1" data={data} searchComponent={search} updateFilter={filter} />)
    // make sure the search component got called with its onFilter prop
    expect(search.mock.calls[0][0].onFilter).toEqual(filter)
  })
})
