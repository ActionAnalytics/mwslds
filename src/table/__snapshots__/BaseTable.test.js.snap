// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTable renders 1`] = `
<div
  className="react-bs-table-container"
>
  <div
    className="react-bs-table react-bs-table-bordered"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "height": "100%",
        "maxHeight": undefined,
      }
    }
  >
    <div
      className="react-bs-container-header table-header-wrapper"
    >
      <table
        className="table table-hover table-bordered"
      >
        <colgroup>
          <col
            key="0"
            style={
              Object {
                "display": null,
              }
            }
          />
        </colgroup>
        <thead>
          <tr
            key="0"
          >
            <th
              className=""
              data-is-only-head={false}
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              title="No Data"
            >
              No Data
              <div
                onClick={[Function]}
              />
            </th>
          </tr>
        </thead>
      </table>
    </div>
    <div
      className="react-bs-container-body"
      style={
        Object {
          "height": "100%",
          "maxHeight": undefined,
        }
      }
    >
      <table
        className="table table-bordered"
      >
        <colgroup>
          <col
            key="0"
            style={
              Object {
                "display": null,
              }
            }
          />
        </colgroup>
        <tbody>
          <tr
            className=""
            hidden={false}
            onClick={[Function]}
            onDoubleClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            style={Object {}}
          >
            <td
              className="react-bs-table-no-data"
              colSpan={1}
              data-toggle="collapse"
            >
              There is no data to display
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    className="react-bs-table-pagination"
  >
    <div
      className="row"
      style={
        Object {
          "marginTop": 15,
        }
      }
    >
      <div
        className="col-md-6 col-xs-6 col-sm-6 col-lg-6"
        key="paging-left"
      >
        <span>
          Showing rows 
          0
           to 
          0
           of 
          0
        </span>
        <span
          className="dropdown   react-bs-table-sizePerPage-dropdown"
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="btn btn-default btn-secondary dropdown-toggle"
            data-toggle="dropdown"
            id="pageDropDown"
            onBlur={[Function]}
            onClick={[Function]}
          >
            10
            <span>
               
              <span
                className="caret"
              />
            </span>
          </button>
          <ul
            aria-labelledby="pageDropDown"
            className="dropdown-menu "
            role="menu"
          >
            <li
              className="dropdown-item"
              key="10"
              role="presentation"
            >
              <a
                data-page={10}
                href="#"
                onMouseDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                10
              </a>
            </li>
            <li
              className="dropdown-item"
              key="25"
              role="presentation"
            >
              <a
                data-page={25}
                href="#"
                onMouseDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                25
              </a>
            </li>
            <li
              className="dropdown-item"
              key="30"
              role="presentation"
            >
              <a
                data-page={30}
                href="#"
                onMouseDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                30
              </a>
            </li>
            <li
              className="dropdown-item"
              key="50"
              role="presentation"
            >
              <a
                data-page={50}
                href="#"
                onMouseDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                50
              </a>
            </li>
          </ul>
        </span>
      </div>
      <div
        className="col-md-6 col-xs-6 col-sm-6 col-lg-6"
        key="paging-right"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <ul
          className="react-bootstrap-table-page-btns-ul pagination"
        />
      </div>
    </div>
  </div>
  <div
    className="s-alert-wrapper"
  />
</div>
`;

exports[`BaseTable renders column labels 1`] = `
<div
  className="react-bs-table-container"
>
  <div
    className="react-bs-table react-bs-table-bordered"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "height": "100%",
        "maxHeight": undefined,
      }
    }
  >
    <div
      className="react-bs-container-header table-header-wrapper"
    >
      <table
        className="table table-hover table-bordered"
      >
        <colgroup>
          <col
            key="0"
            style={
              Object {
                "display": null,
              }
            }
          />
          <col
            key="1"
            style={
              Object {
                "display": null,
              }
            }
          />
          <col
            key="2"
            style={
              Object {
                "display": null,
              }
            }
          />
        </colgroup>
        <thead>
          <tr
            key="0"
          >
            <th
              className=""
              data-is-only-head={false}
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              title="first"
            >
              first
              <div
                onClick={[Function]}
              />
            </th>
            <th
              className=""
              data-is-only-head={false}
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              title="second"
            >
              second
              <div
                onClick={[Function]}
              />
            </th>
            <th
              className=""
              data-is-only-head={false}
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              title="third"
            >
              third
              <div
                onClick={[Function]}
              />
            </th>
          </tr>
        </thead>
      </table>
    </div>
    <div
      className="react-bs-container-body"
      style={
        Object {
          "height": "100%",
          "maxHeight": undefined,
        }
      }
    >
      <table
        className="table table-bordered"
      >
        <colgroup>
          <col
            key="0"
            style={
              Object {
                "display": null,
              }
            }
          />
          <col
            key="1"
            style={
              Object {
                "display": null,
              }
            }
          />
          <col
            key="2"
            style={
              Object {
                "display": null,
              }
            }
          />
        </colgroup>
        <tbody>
          <tr
            className=""
            hidden={false}
            onClick={[Function]}
            onDoubleClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            style={Object {}}
          >
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="1"
              title={null}
            >
              aa
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="2"
              title={null}
            >
              ab
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="3"
              title={null}
            >
              ac
            </td>
          </tr>
          <tr
            className=""
            hidden={false}
            onClick={[Function]}
            onDoubleClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            style={Object {}}
          >
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="4"
              title={null}
            >
              ba
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="5"
              title={null}
            >
              bb
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="6"
              title={null}
            >
              bc
            </td>
          </tr>
          <tr
            className=""
            hidden={false}
            onClick={[Function]}
            onDoubleClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            style={Object {}}
          >
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="7"
              title={null}
            >
              ca
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="8"
              title={null}
            >
              cb
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="9"
              title={null}
            >
              cc
            </td>
          </tr>
          <tr
            className=""
            hidden={false}
            onClick={[Function]}
            onDoubleClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            style={Object {}}
          >
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="10"
              title={null}
            >
              da
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="11"
              title={null}
            >
              db
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="12"
              title={null}
            >
              dc
            </td>
          </tr>
          <tr
            className=""
            hidden={false}
            onClick={[Function]}
            onDoubleClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            style={Object {}}
          >
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="13"
              title={null}
            >
              ea
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="14"
              title={null}
            >
              eb
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="15"
              title={null}
            >
              ec
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    className="react-bs-table-pagination"
  >
    <div
      className="row"
      style={
        Object {
          "marginTop": 15,
        }
      }
    >
      <div
        className="col-md-6 col-xs-6 col-sm-6 col-lg-6"
        key="paging-left"
      >
        <span>
          Showing rows 
          1
           to 
          5
           of 
          5
        </span>
        <span
          className="dropdown   react-bs-table-sizePerPage-dropdown"
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="btn btn-default btn-secondary dropdown-toggle"
            data-toggle="dropdown"
            id="pageDropDown"
            onBlur={[Function]}
            onClick={[Function]}
          >
            10
            <span>
               
              <span
                className="caret"
              />
            </span>
          </button>
          <ul
            aria-labelledby="pageDropDown"
            className="dropdown-menu "
            role="menu"
          >
            <li
              className="dropdown-item"
              key="10"
              role="presentation"
            >
              <a
                data-page={10}
                href="#"
                onMouseDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                10
              </a>
            </li>
            <li
              className="dropdown-item"
              key="25"
              role="presentation"
            >
              <a
                data-page={25}
                href="#"
                onMouseDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                25
              </a>
            </li>
            <li
              className="dropdown-item"
              key="30"
              role="presentation"
            >
              <a
                data-page={30}
                href="#"
                onMouseDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                30
              </a>
            </li>
            <li
              className="dropdown-item"
              key="50"
              role="presentation"
            >
              <a
                data-page={50}
                href="#"
                onMouseDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                50
              </a>
            </li>
          </ul>
        </span>
      </div>
      <div
        className="col-md-6 col-xs-6 col-sm-6 col-lg-6"
        key="paging-right"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <ul
          className="react-bootstrap-table-page-btns-ul pagination"
        >
          <li
            className="active page-item"
            title="1"
          >
            <a
              className="page-link"
              href="#"
              onClick={[Function]}
            >
              1
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="s-alert-wrapper"
  />
</div>
`;

exports[`BaseTable renders expand indicator column 1`] = `
<div
  className="react-bs-table-container"
>
  <div
    className="react-bs-table react-bs-table-bordered"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "height": "100%",
        "maxHeight": undefined,
      }
    }
  >
    <div
      className="react-bs-container-header table-header-wrapper"
    >
      <table
        className="table table-hover table-bordered"
      >
        <colgroup>
          <col
            key="expand-col"
            style={
              Object {
                "minWidth": "30px",
                "width": "30px",
              }
            }
          />
          <col
            key="0"
            style={
              Object {
                "display": null,
              }
            }
          />
          <col
            key="1"
            style={
              Object {
                "display": null,
              }
            }
          />
          <col
            key="2"
            style={
              Object {
                "display": null,
              }
            }
          />
        </colgroup>
        <thead>
          <tr
            key="0"
          >
            <th
              className="react-bs-table-expand-cell"
              data-is-only-head={false}
              rowSpan={1}
              style={
                Object {
                  "textAlign": "center",
                }
              }
            />
            <th
              className=""
              data-is-only-head={false}
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              title="Col 1"
            >
              Col 1
              <div
                onClick={[Function]}
              />
            </th>
            <th
              className=""
              data-is-only-head={false}
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              title="Col 2"
            >
              Col 2
              <div
                onClick={[Function]}
              />
            </th>
            <th
              className=""
              data-is-only-head={false}
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              title="Col 3"
            >
              Col 3
              <div
                onClick={[Function]}
              />
            </th>
          </tr>
        </thead>
      </table>
    </div>
    <div
      className="react-bs-container-body"
      style={
        Object {
          "height": "100%",
          "maxHeight": undefined,
        }
      }
    >
      <table
        className="table table-bordered"
      >
        <colgroup>
          <col
            key="expand-col"
            style={
              Object {
                "minWidth": "30px",
                "width": "30px",
              }
            }
          />
          <col
            key="0"
            style={
              Object {
                "display": null,
              }
            }
          />
          <col
            key="1"
            style={
              Object {
                "display": null,
              }
            }
          />
          <col
            key="2"
            style={
              Object {
                "display": null,
              }
            }
          />
        </colgroup>
        <tbody>
          <tr
            className=""
            hidden={false}
            onClick={[Function]}
            onDoubleClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            style={Object {}}
          >
            <td
              className="react-bs-table-expand-cell"
              onClick={[Function]}
            >
              <div
                className="text-center text-clickable"
              >
                +
              </div>
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="1"
              title={null}
            >
              aa
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="2"
              title={null}
            >
              ab
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="3"
              title={null}
            >
              ac
            </td>
          </tr>
          <tr
            className=""
            hidden={true}
            style={
              Object {
                "backgroundColor": undefined,
              }
            }
            width="100%"
          >
            <td
              colSpan={4}
            >
              <div />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    className="react-bs-table-pagination"
  >
    <div
      className="row"
      style={
        Object {
          "marginTop": 15,
        }
      }
    >
      <div
        className="col-md-6 col-xs-6 col-sm-6 col-lg-6"
        key="paging-left"
      >
        <span>
          Showing rows 
          1
           to 
          1
           of 
          1
        </span>
        <span
          className="dropdown   react-bs-table-sizePerPage-dropdown"
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="btn btn-default btn-secondary dropdown-toggle"
            data-toggle="dropdown"
            id="pageDropDown"
            onBlur={[Function]}
            onClick={[Function]}
          >
            10
            <span>
               
              <span
                className="caret"
              />
            </span>
          </button>
          <ul
            aria-labelledby="pageDropDown"
            className="dropdown-menu "
            role="menu"
          >
            <li
              className="dropdown-item"
              key="10"
              role="presentation"
            >
              <a
                data-page={10}
                href="#"
                onMouseDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                10
              </a>
            </li>
            <li
              className="dropdown-item"
              key="25"
              role="presentation"
            >
              <a
                data-page={25}
                href="#"
                onMouseDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                25
              </a>
            </li>
            <li
              className="dropdown-item"
              key="30"
              role="presentation"
            >
              <a
                data-page={30}
                href="#"
                onMouseDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                30
              </a>
            </li>
            <li
              className="dropdown-item"
              key="50"
              role="presentation"
            >
              <a
                data-page={50}
                href="#"
                onMouseDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                50
              </a>
            </li>
          </ul>
        </span>
      </div>
      <div
        className="col-md-6 col-xs-6 col-sm-6 col-lg-6"
        key="paging-right"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <ul
          className="react-bootstrap-table-page-btns-ul pagination"
        >
          <li
            className="active page-item"
            title="1"
          >
            <a
              className="page-link"
              href="#"
              onClick={[Function]}
            >
              1
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="s-alert-wrapper"
  />
</div>
`;

exports[`BaseTable renders with data 1`] = `
<div
  className="react-bs-table-container"
>
  <div
    className="react-bs-table react-bs-table-bordered"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "height": "100%",
        "maxHeight": undefined,
      }
    }
  >
    <div
      className="react-bs-container-header table-header-wrapper"
    >
      <table
        className="table table-hover table-bordered"
      >
        <colgroup>
          <col
            key="0"
            style={
              Object {
                "display": null,
              }
            }
          />
          <col
            key="1"
            style={
              Object {
                "display": null,
              }
            }
          />
          <col
            key="2"
            style={
              Object {
                "display": null,
              }
            }
          />
        </colgroup>
        <thead>
          <tr
            key="0"
          >
            <th
              className=""
              data-is-only-head={false}
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              title="Col 1"
            >
              Col 1
              <div
                onClick={[Function]}
              />
            </th>
            <th
              className=""
              data-is-only-head={false}
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              title="Col 2"
            >
              Col 2
              <div
                onClick={[Function]}
              />
            </th>
            <th
              className=""
              data-is-only-head={false}
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              title="Col 3"
            >
              Col 3
              <div
                onClick={[Function]}
              />
            </th>
          </tr>
        </thead>
      </table>
    </div>
    <div
      className="react-bs-container-body"
      style={
        Object {
          "height": "100%",
          "maxHeight": undefined,
        }
      }
    >
      <table
        className="table table-bordered"
      >
        <colgroup>
          <col
            key="0"
            style={
              Object {
                "display": null,
              }
            }
          />
          <col
            key="1"
            style={
              Object {
                "display": null,
              }
            }
          />
          <col
            key="2"
            style={
              Object {
                "display": null,
              }
            }
          />
        </colgroup>
        <tbody>
          <tr
            className=""
            hidden={false}
            onClick={[Function]}
            onDoubleClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            style={Object {}}
          >
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="1"
              title={null}
            >
              aa
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="2"
              title={null}
            >
              ab
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="3"
              title={null}
            >
              ac
            </td>
          </tr>
          <tr
            className=""
            hidden={false}
            onClick={[Function]}
            onDoubleClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            style={Object {}}
          >
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="4"
              title={null}
            >
              ba
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="5"
              title={null}
            >
              bb
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="6"
              title={null}
            >
              bc
            </td>
          </tr>
          <tr
            className=""
            hidden={false}
            onClick={[Function]}
            onDoubleClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            style={Object {}}
          >
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="7"
              title={null}
            >
              ca
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="8"
              title={null}
            >
              cb
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="9"
              title={null}
            >
              cc
            </td>
          </tr>
          <tr
            className=""
            hidden={false}
            onClick={[Function]}
            onDoubleClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            style={Object {}}
          >
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="10"
              title={null}
            >
              da
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="11"
              title={null}
            >
              db
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="12"
              title={null}
            >
              dc
            </td>
          </tr>
          <tr
            className=""
            hidden={false}
            onClick={[Function]}
            onDoubleClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            style={Object {}}
          >
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="13"
              title={null}
            >
              ea
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="14"
              title={null}
            >
              eb
            </td>
            <td
              className=""
              onClick={[Function]}
              style={
                Object {
                  "display": null,
                  "textAlign": "left",
                }
              }
              tabIndex="15"
              title={null}
            >
              ec
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    className="react-bs-table-pagination"
  >
    <div
      className="row"
      style={
        Object {
          "marginTop": 15,
        }
      }
    >
      <div
        className="col-md-6 col-xs-6 col-sm-6 col-lg-6"
        key="paging-left"
      >
        <span>
          Showing rows 
          1
           to 
          5
           of 
          5
        </span>
        <span
          className="dropdown   react-bs-table-sizePerPage-dropdown"
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="btn btn-default btn-secondary dropdown-toggle"
            data-toggle="dropdown"
            id="pageDropDown"
            onBlur={[Function]}
            onClick={[Function]}
          >
            10
            <span>
               
              <span
                className="caret"
              />
            </span>
          </button>
          <ul
            aria-labelledby="pageDropDown"
            className="dropdown-menu "
            role="menu"
          >
            <li
              className="dropdown-item"
              key="10"
              role="presentation"
            >
              <a
                data-page={10}
                href="#"
                onMouseDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                10
              </a>
            </li>
            <li
              className="dropdown-item"
              key="25"
              role="presentation"
            >
              <a
                data-page={25}
                href="#"
                onMouseDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                25
              </a>
            </li>
            <li
              className="dropdown-item"
              key="30"
              role="presentation"
            >
              <a
                data-page={30}
                href="#"
                onMouseDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                30
              </a>
            </li>
            <li
              className="dropdown-item"
              key="50"
              role="presentation"
            >
              <a
                data-page={50}
                href="#"
                onMouseDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                50
              </a>
            </li>
          </ul>
        </span>
      </div>
      <div
        className="col-md-6 col-xs-6 col-sm-6 col-lg-6"
        key="paging-right"
        style={
          Object {
            "display": "block",
          }
        }
      >
        <ul
          className="react-bootstrap-table-page-btns-ul pagination"
        >
          <li
            className="active page-item"
            title="1"
          >
            <a
              className="page-link"
              href="#"
              onClick={[Function]}
            >
              1
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="s-alert-wrapper"
  />
</div>
`;
